dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java

```diff
@@ -192,24 +192,47 @@ private String buildDataxJsonFile()
         throws Exception {
         // generate json
         String fileName = String.format("%s/%s_job.json", taskDir, taskProps.getTaskAppId());
+        String json;
 
         Path path = new File(fileName).toPath();
         if (Files.exists(path)) {
             return fileName;
         }
 
-        JSONObject job = new JSONObject();
-        job.put("content", buildDataxJobContentJson());
-        job.put("setting", buildDataxJobSettingJson());
 
-        JSONObject root = new JSONObject();
-        root.put("job", job);
-        root.put("core", buildDataxCoreJson());
 
-        logger.debug("datax job json : {}", root.toString());
+        if (dataXParameters.getCustomConfig() == 1){
+
+            json = dataXParameters.getJson().replaceAll("\\r\\n", "\n");
+
```

Comment:
Variables can also be used in datax job json, which will cause json to be invalid, such as
```
{
  ...
  "reader": {
    "name": "mysqlreader",
    "parameter": {
      "username": "${db_user}",
      "password": "${db_pwd}",
      "encoding": "UTF-8",
      "column": [${tbl_clms}],
      "connection": [
        {
          "jdbcUrl": [
            "jdbc:mysql://${host}/${db}?useUnicode=true&characterEncoding=utf8"
          ],
          "table": [
            ${table_list}
          ]
        }
      ]
    }
  }
  ...
}
```

Reply:
here is about custom variables 
![image](https://user-images.githubusercontent.com/18065113/77021783-afaace00-69c2-11ea-9db7-db09767daee7.png)

Reply:
> here is about custom variables
Sorry, I didn't see it. it's my fault.